tape/src/main/java/com/squareup/tape/QueueElementObserver.java

```diff
@@ -0,0 +1,26 @@
+package com.squareup.tape;
+
+import java.io.IOException;
+
+public interface QueueElementObserver {
+
+  /**
+   * Observe elements flowing through the Tape queue, to make handling decisions.
+   *
+   * @param elementSize - Size of the element being added or read in bytes.
+   * @param maxElementSize - Max size of elements for this queue.
+   * @param fileContentLength - Current size of the content length as per the file header.
+   * @param isAdd - Whether the element is being added. False means element is being read.
+   * @param isInit - Whether this is during queue initialization from an existing file.
```

Comment:
probably need to clarify why init calls back, ie how elements are involved in init